packages/rainbowkit/src/components/ChainModal/ChainModal.tsx

```diff
@@ -104,114 +70,124 @@ export function ChainModal({ onClose, open }: ChainModalProps) {
           )}
           <Box display="flex" flexDirection="column" gap="4" padding="2">
             {switchNetwork ? (
-              chains.map((chain, idx) => {
-                const isCurrentChain = chain.id === activeChain?.id;
-                const switching = chain.id === switchingToChain;
-                const rainbowKitChain = rainbowkitChainsById[chain.id];
-                const chainIconSize: BoxProps['width'] = mobile ? '36' : '28';
-                const chainIconUrl = rainbowKitChain?.iconUrl;
-                const chainIconBackground = rainbowKitChain?.iconBackground;
+              Object.values(rainbowkitChainsById).map(
+                (rainbowKitChain, idx) => {
+                  const chain = chains.find(c => rainbowKitChain.id === c.id);
+                  if (!chain) return null;
 
-                return (
-                  <Fragment key={chain.id}>
-                    <MenuButton
-                      currentlySelected={isCurrentChain}
-                      onClick={
-                        isCurrentChain
-                          ? undefined
-                          : () => {
-                              setSwitchingToChain(chain.id);
-                              switchNetwork(chain.id);
-                            }
-                      }
-                      testId={`chain-option-${chain.id}`}
-                    >
-                      <Box fontFamily="body" fontSize="16" fontWeight="bold">
-                        <Box
-                          alignItems="center"
-                          display="flex"
-                          flexDirection="row"
-                          justifyContent="space-between"
-                        >
+                  let status = 'idle';
+                  if (chain.id === pendingChainId) status = 'switching';
+                  if (chain.id === activeChain?.id) status = 'current';
+
+                  const isCurrentChain = status === 'current';
+                  const switching = status === 'switching';
```

Comment:
this lil "finite state machine" aims to solve this issue
<img width="376" alt="Screenshot 2023-04-08 at 18 43 01" src="https://user-images.githubusercontent.com/6232729/230743842-477f4267-673e-4715-842f-c63373314c51.png">

before:
https://user-images.githubusercontent.com/6232729/230744078-b69f0ee2-09bc-4cfc-a490-cdef78169735.mov

after:

https://user-images.githubusercontent.com/6232729/230744084-d8a432f3-62c2-46d9-a485-92458813ec2a.mov

check how in this pr implementation, the double state won't happen, but the "connect" text color is white (?) any idea why is that? (it's only white in the switching -> connected -> close, when opening again it's the right color)
<img width="495" alt="Screenshot 2023-04-08 at 18 53 24" src="https://user-images.githubusercontent.com/6232729/230744125-394d3910-6170-4e60-b952-5b6f07a943b2.png">

Reply:
> check how in this pr implementation, the double state won't happen, but the "connect" text color is white (?) any idea why is that? (it's only white in the switching -> connected -> close, when opening again it's the right color)

found that's due to the `<MenuButton />` having a bg color transition, not sure how to "fix" that, or if it should be fixed